### PR TITLE
Fix flawed require hook node_modules check

### PR DESCRIPTION
**Pull request by jmm**
The require hook currently does a naive test for presence of substring `node_modules` in the path, which results in a false positive in case of something like `require('./not_node_modules/whatever')`.

Repro:
```sh
$ git clone https://github.com/jmm/babel-require-hook-not-node_modules-repro
$ cd babel-require-hook-not-node_modules-repro
$ npm install
$ node ./run.js
```

Is using `path.sep` appropriate here, or are the separators normalized to something (presumably `/`)?

I took a stab at figuring out where to locate test files.

### CODE REVIEW

File: test/core/require-hook.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var chai        = require("chai");
```
Comment (sebmck) [inline test/core/require-hook.js]: Might be better to put this test [here](https://github.com/babel/babel/tree/master/test/core/fixtures/bin/babel-node) and use a more robust test than just a bare arrow function which will pass if it's successfully ignored and the target environment supports arrow functions.

**Comment by jmm** [comment]
> Might be better to put this test here

Are you saying move the fixture dir I created to  `test/core/fixtures/bin/babel-node/require-hook`? And leave the test file where it is? Sorry if I'm misunderstanding.

> and use a more robust test than just a bare arrow function

I had the same thought and I was hoping you'd have a good recommendation.

**Comment by sebmck** [comment]
> Are you saying move the fixture dir I created to  `test/core/fixtures/bin/babel-node/require-hook`? And leave the test file where it is? Sorry if I'm misunderstanding.

As in, add it as a test condition in [this](https://github.com/babel/babel/tree/master/test/core/fixtures/bin/babel-node/require) fixture directory. You'll just need to create a `not_node_modules.js` file [here](https://github.com/babel/babel/tree/master/test/core/fixtures/bin/babel-node/require/in-files) and `require` it from [foo2.js](https://github.com/babel/babel/blob/master/test/core/fixtures/bin/babel-node/require/in-files/foo2.js).

> I had the same thought and I was hoping you'd have a good recommendation.

JSX is probably a good bet since it's likely to never come to a native environment.

**Comment by jmm** [comment]
@sebmck 
> As in, add it as a test condition in this fixture directory. You'll just need to create a `not_node_modules.js` file here and `require/import` it from foo2.js.

Ok, thanks. FWIW that produces a pretty ugly error message (see below).

> JSX is probably a good bet since it's likely to never come to a native environment.

Ok, thanks.

Please see https://github.com/babel/babel/compare/master...jmm:node_modules-robust2.

```
1) bin/babel-node require:
   Error: /somedir/babel/babel/packages/babel-cli/bin/babel-node foo2: stderr:
   "/somedir/babel/babel/test/core/tmp/not_node_modules.jsx:8\n  <Some jsx=
   \"element\" />;\n  ^\nSyntaxError: Unexpected token <\n    at Module._compile
   (module.js:439:25)\n    at Module._extensions..js (module.js:474:10)\n    at
   Object.require.extensions.(anonymous function) [as .jsx]
   (/somedir/babel/babel/lib/babel/api/register/node.js:157:7)\n    at
   Module.load (module.js:356:32)\n    at Function.Module._load
   (module.js:312:12)\n    at Module.require (module.js:364:17)\n    at require
   (module.js:380:17)\n    at Object.<anonymous>
   (/somedir/babel/babel/test/core/tmp/foo2.js:1:16)\n    at Module._compile
   (module.js:456:26)\n    at normalLoader
   (/somedir/babel/babel/lib/babel/api/register/node.js:146:5)"
      at assertTest (/somedir/babel/babel/test/core/bin.js:45:11)
      at ChildProcess.<anonymous> (/somedir/babel/babel/test/core/bin.js:94:9)
      at ChildProcess.emit (events.js:98:17)
      at maybeClose (child_process.js:756:16)
      at Process.ChildProcess._handle.onexit (child_process.js:823:5)
```

**Comment by sebmck** [comment]
Looks like the ignore check isn't working.

**Comment by jmm** [comment]
@sebmck 
> Looks like the ignore check isn't working.

Sorry, what do you mean?

In case I was unclear, the error message I posted is what is displayed when the test you suggested fails, as it is intended to in https://github.com/babel/babel/commit/227f4600bf8f4ef96ba44ddf7e189aafdca2f2f0. No one will ever see it unless this breaks in the future, in which case they'll be able to figure it out, but it's not a very user-friendly error message.

**Comment by sebmck** [comment]
Oh right, yeah that's fine.

**Comment by sebmck** [comment]
Thanks!

**Pull request merged**